tests/backup/backup_upgrade_test.go

```diff
@@ -0,0 +1,186 @@
+package tests
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/pborman/uuid"
+	api "github.com/portworx/px-backup-api/pkg/apis/v1"
+	"github.com/portworx/torpedo/drivers/backup"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	. "github.com/portworx/torpedo/tests"
+)
+
+// StorkUpgradeWithBackup validates backups with stork upgrade
+var _ = Describe("{StorkUpgradeWithBackup}", func() {
+	var (
+		contexts             []*scheduler.Context
+		appContexts          []*scheduler.Context
+		backupLocationName   string
+		backupLocationUID    string
+		cloudCredUID         string
+		bkpNamespaces        []string
+		scheduleNames        []string
+		cloudAccountName     string
+		backupName           string
+		scheduleName         string
+		schBackupName        string
+		schPolicyUid         string
+		upgradeStorkImageStr string
+		clusterUid           string
+		clusterStatus        api.ClusterInfo_StatusInfo_Status
+	)
+	var testrailID = 58023 // testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/58023
+	labelSelectors := make(map[string]string)
+	cloudCredUIDMap := make(map[string]string)
+	backupLocationMap := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	timeStamp := strconv.Itoa(int(time.Now().Unix()))
+	periodicPolicyName := fmt.Sprintf("%s-%s", "periodic", timeStamp)
+	upgradeStorkImageStr = getEnv(upgradeStorkImage, "openstorage/stork:23.2.0")
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("StorkUpgradeWithBackup", "Validates the scheduled backups and creation of new backup after stork upgrade", nil, testrailID)
+		log.Info("Application installation")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+
+	It("StorkUpgradeWithBackup", func() {
+		Step("Validate deployed applications", func() {
+			ValidateApplications(contexts)
+		})
+		providers := getProviders()
+		Step("Adding Cloud Account", func() {
```

Comment:
Why can't w combine both Cloud Account and Backup Location in a single step?

Reply:
Done.